src/components/SelectionList/index.tsx

```diff
@@ -62,6 +62,10 @@ function SelectionList<TItem extends ListItem>({onScroll, ...props}: SelectionLi
         if (!isScreenTouched) {
             return;
         }
+
+        if (!shouldHideKeyboardOnScroll) {
+            return;
+        }
```

Comment:
These two can be combined (and comment adjusted):
```suggestion
        // Only dismiss the keyboard whenever the user scrolls the screen or `shouldHideKeyboardOnScroll` is true
        if (!isScreenTouched || !shouldHideKeyboardOnScroll) {
            return;
        }
```